internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -931,6 +997,42 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	return resourceRead(ctx, d, meta)
 }
 
+func handlePinnedFCVUpdate(ctx context.Context, connV2 *admin.APIClient, projectID, clusterName string, d *schema.ResourceData, timeout time.Duration) diag.Diagnostics {
+	if d.HasChange("pinned_fcv") {
+		if pinnedFCVBlock, ok := d.Get("pinned_fcv").([]any); ok && len(pinnedFCVBlock) > 0 {
+			if diags := pinFCV(ctx, connV2, projectID, clusterName, pinnedFCVBlock[0]); diags.HasError() {
+				return diags
+			}
+		} else {
+			// pinned_fcv has been removed from the config so unpin method is called
```

Comment:
similarly to prev. comment, is it possible to pick var names so comments are not needed? (maybe not)